### ISSUE TITLE
 Missing type definition `hot` in `BaseEditor`, `rootDocument`, `rootWindow`, `rootElement` in `Core`

### ISSUE THREAD

**Issue by imageslr** [handsontable/handsontable#6168]
I'm implementing a custom editor following the tutorial:
``` js
export default class AditEditor extends Handsontable.editors.TextEditor
  createElements() {
    super.createElements();
    this.DIV = this.hot.rootDocument.createElement('DIV');
  }
}
```

In typescript this will cause a problem: `type “AditEditor” doesn't have property “hot”`.

If I modify it to `this.instance.rootDocument`, the error would be ``type “Core” doesn't have property “rootDocument”`.

**Comment by wojciechczerniak** [comment]
@imageslr May I know why you've closed the issue? If you found a solution it might be helpful for others who will have this problem in the future

**Comment by imageslr** [comment]
@wojciechczerniak Hello, I have not found a solution beside adding declaration in `handsontable.d.ts`. I thought `rootDocument`/ `rootWindow` are properties that `Handsontable` doesn't want to export, so I closed the issue. 

I found `Handsontable.rootDocument` is same with global variable `document`. So I can access `document` (but I still don't know why there is a `rootDocument` in handsontable instance).

Feel free to reopen the issue if this is indeed a problem!. Thank you for reviewing!

**Comment by AMBudnik** [comment]
Reopening.

**Comment by AMBudnik** [comment]
Hey @imageslr 

we've just released version `7.2.0` where we improved the TypeScript file.

Thank you very much.
